src/stores/store/keys/hashicorp/hashicorp.go

```diff
@@ -52,11 +58,16 @@ func (s *Store) Create(_ context.Context, id string, alg *entities2.Algorithm, a
 		s.logger.With("id", id).WithError(err).Error(errMessage)
 		return nil, errors.FromError(err).SetMessage(errMessage)
 	}
-
 	return parseAPISecretToKey(res)
 }
 
 func (s *Store) Import(_ context.Context, id string, privKey []byte, alg *entities2.Algorithm, attr *entities.Attributes) (*entities.Key, error) {
+	if !s.isSupportedAlgo(alg) {
+		errMessage := "invalid or not supported elliptic curve and signing algorithm for Hashicorp key import"
+		s.logger.With("elliptic_curve", alg.EllipticCurve, "signing_algorithm", alg.Type).Error(errMessage)
+		return nil, errors.NotSupportedError(errMessage)
```

Comment:
Are we sure we want to return `NotSupported` ? I think there is a difference between these 3 error types:
- Not Implemented means that the pair curve/algo exists but it's not implemented for this store
- Not Supported means that the pair curve/algo exists but this store does not support it
- Invalid Parameter means the pair curve/algo does not exist (invalid).

Here it seems we want InvalidParameter